File: lessons/5.3/1-test.php
<?php

require 'vendor/autoload.php';
require 'inc/header.php';

echo "<pre>";
echo "Setting up mongo client\n";

$client = new MongoDB\Client(
    'mongodb://localhost:27017/cooker'
);

echo "Connecting to database ...\n";

$db = $client->cooker;

echo "Connected Successfully!\n";

$database_name = $db->getDatabaseName();

echo "The database name is: {$database_name}\n";

echo "</pre>";

require 'inc/footer.php';

File: lessons/5.4/1-test.go
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)

func main() {

	fmt.Println("Setting up mongo client")

	client, err := mongo.NewClient(options.Client().ApplyURI("mongodb://localhost:27017"))

	if err != nil {
		log.Fatal(err)
	}

	fmt.Println("Connecting to database ...")

	context, _ := context.WithTimeout(context.Background(), 10*time.Second)
	err = client.Connect(context)

	if err != nil {
		log.Fatal(err)
	}

	fmt.Println("Connected Successfully!")

	defer client.Disconnect(context)

	fmt.Println("List databases ...")

	databases, err := client.ListDatabaseNames(context, bson.M{})

	if err != nil {
		log.Fatal(err)
	}

	fmt.Println(databases)

}
